includes/admin/meta-boxes/views/html-variation-admin.php

```diff
@@ -153,99 +179,109 @@
-							<label><?php echo __( 'Weight', 'woocommerce' ) . ' (' . esc_html( get_option( 'woocommerce_weight_unit' ) ) . ')'; ?> <?php echo wc_help_tip( __( 'Enter a weight for this variation or leave blank to use the parent product weight.', 'woocommerce' ) ); ?></a></label>
-							<input type="text" size="5" name="variable_weight[<?php echo $loop; ?>]" value="<?php if ( isset( $_weight ) ) echo esc_attr( $_weight ); ?>" placeholder="<?php echo esc_attr( $parent_data['weight'] ); ?>" class="wc_input_decimal" />
-						</p>
-					<?php else : ?>
-						<p>&nbsp;</p>
-					<?php endif; ?>
-					<?php if ( wc_product_dimensions_enabled() ) : ?>
-						<p class="form-row dimensions_field hide_if_variation_virtual form-row-last">
-							<label for="product_length"><?php echo __( 'Dimensions (L&times;W&times;H)', 'woocommerce' ) . ' (' . esc_html( get_option( 'woocommerce_dimension_unit' ) ) . ')'; ?></label>
-							<input id="product_length" class="input-text wc_input_decimal" size="6" type="text" name="variable_length[<?php echo $loop; ?>]" value="<?php if ( isset( $_length ) ) echo esc_attr( $_length ); ?>" placeholder="<?php echo esc_attr( $parent_data['length'] ); ?>" />
-							<input class="input-text wc_input_decimal" size="6" type="text" name="variable_width[<?php echo $loop; ?>]" value="<?php if ( isset( $_width ) ) echo esc_attr( $_width ); ?>" placeholder="<?php echo esc_attr( $parent_data['width'] ); ?>" />
-							<input class="input-text wc_input_decimal last" size="6" type="text" name="variable_height[<?php echo $loop; ?>]" value="<?php if ( isset( $_height ) ) echo esc_attr( $_height ); ?>" placeholder="<?php echo esc_attr( $parent_data['height'] ); ?>" />
-						</p>
-					<?php else : ?>
-						<p>&nbsp;</p>
-					<?php endif; ?>
+					if ( wc_product_weight_enabled() ) {
+						woocommerce_wp_text_input( array(
+							'id'            => "variable_weight{$loop}",
+							'name'          => "variable_weight[{$loop}]",
+							'value'         => wc_format_localized_decimal( $variation_object->get_weight() ), // @todo need method to get raw unfiltered values in admin.
+							'placeholder'   => wc_format_localized_decimal( $product_object->get_weight() ),
+							'label'         => __( 'Weight', 'woocommerce' ) . ' (' . get_option( 'woocommerce_weight_unit' ) . ')',
+							'desc_tip'      => true,
+							'description'   => __( 'Weight in decimal form', 'woocommerce' ),
+							'type'          => 'text',
+							'data_type'     => 'decimal',
+							'wrapper_class' => 'form-row form-row-first hide_if_variation_virtual',
+						) );
+					}
+
+					if ( wc_product_dimensions_enabled() ) {
+						?><p class="form-field form-row dimensions_field hide_if_variation_virtual form-row-last">
+							<label for="product_length"><?php echo __( 'Dimensions', 'woocommerce' ) . ' (' . get_option( 'woocommerce_dimension_unit' ) . ')'; ?></label>
+							<?php echo wc_help_tip( __( 'LxWxH in decimal form', 'woocommerce' ) ); ?>
```

Comment:
Since this is displayed in a help tip/on hover, can we write out length x width x height?